doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1257,409 +1261,344 @@ Managing the local context
-      the hypothesis is named after this introduction pattern (in particular, if
-      :n:`simple_intropattern` is :n:`@ident`, the tactic behaves like
-      :n:`assert (@ident : @type)`). If :n:`simple_intropattern` is an action
-      introduction pattern, the tactic behaves like :n:`assert @type` followed by
-      the action done by this introduction pattern.
-
-   .. tacv:: assert @type as @simple_intropattern by @tactic
+   .. exn:: The term "@type" has type "@type__1" which should be Set, Prop or Type.
 
-      This combines the two previous variants of :tacn:`assert`.
+      Occurs when the argument :n:`@type` (in the first form) or :n:`@one_term`
+      (in the third form) is not of type :g:`Prop`, :g:`Set` nor :g:`Type`.
 
-   .. tacv:: assert (@ident := @term)
-
-      This behaves as :n:`assert (@ident : @type) by exact @term` where
-      :token:`type` is the type of :token:`term`. This is equivalent to using
-      :tacn:`pose proof`. If the head of term is :token:`ident`, the tactic
-      behaves as :tacn:`specialize`.
-
-      .. exn:: Variable @ident is already declared.
-         :undocumented:
+   .. exn:: Proof is not complete.
+      :name: Proof is not complete. (assert)
 
-.. tacv:: eassert @type as @simple_intropattern by @tactic
-   :name: eassert
+      :n:`@ltac_expr3` was not able to prove the new hypothesis.
 
-   While the different variants of :tacn:`assert` expect that no existential
-   variables are generated by the tactic, :tacn:`eassert` removes this constraint.
-   This lets you avoid specifying the asserted statement completely before starting
-   to prove it.
+   .. tacn:: eassert ( @ident : @type ) {? by @ltac_expr3 }
+             eassert ( @ident := @term )
+             eassert @one_term {? @as_ipat } {? by @ltac_expr3 }
+      :name: eassert; _; _
 
-.. tacv:: pose proof @term {? as @simple_intropattern}
-   :name: pose proof
+      Unlike :tacn:`assert`, the :n:`@type`\s, :n:`@term`\s and :n:`@one_term`\s in
+      :tacn:`eassert` may contain :gdef:`holes <hole>`, denoted by :n:`_`,
+      for which the tactic will create existential variables.  This lets you
+      avoid specifying the asserted statement completely before starting to
+      prove it.
 
-   This tactic behaves like :n:`assert @type {? as @simple_intropattern} by exact @term`
-   where :token:`type` is the type of :token:`term`. In particular,
-   :n:`pose proof @term as @ident` behaves as :n:`assert (@ident := @term)`
-   and :n:`pose proof @term as @simple_intropattern` is the same as applying the
-   :token:`simple_intropattern` to :token:`term`.
+.. tacn:: enough ( @ident : @type ) {? by @ltac_expr3 }
+          enough @one_term {? @as_ipat } {? by @ltac_expr3 }
+   :name: enough; _
 
-.. tacv:: epose proof @term {? as @simple_intropattern}
-   :name: epose proof
+   Adds a new hypothesis to the current subgoal and a new subgoal to prove the
+   hypothesis.
```

Comment:
I didn't mean that we should not talk about the location on the new subgoal at all, though :grin: What about:
```suggestion
   hypothesis (*after* the current subgoal).
```

Reply:
How about:

   Adds a new hypothesis to the current subgoal and a new subgoal **after it**
   to prove the hypothesis.

I could also say "immediately after" because "after" is not very specific, e.g. it could mean adding after the last of N subgoals.

Reply:
> Adds a new hypothesis to the current subgoal and a new subgoal **after it**
> to prove the hypothesis.

LGTM

> I could also say "immediately after" because "after" is not very specific, e.g. it could mean adding after the last of N subgoals.

IMHO there is no risk of misinterpretation since a tactic cannot insert goals anywhere else than the goal it was applied on (and it is not the role of the documentation of individual tactics to remind that). So "after" seems clear enough.